testcases/parser/visitor.go

```diff
@@ -711,7 +727,12 @@ func (v *TestCaseVisitor) VisitLiteral(ctx *baseparser.LiteralContext) interface
 	}
 
 	if ctx.NullLiteral() != nil {
-		return expr.NewNullLiteral(v.getLiteralTypeInContext())
+		nullType := v.getLiteralTypeInContext()
+		if nullType == nil {
+			// Use a dummy type for null literal. This happens in AggregateFuncCall context, where type is not set
```

Comment:
Could the aggregate type be pulled (or pushed down) from the parent's context?

Reply:
2 out of 3 formats we push down the type. In one case, code is getting messy to push down for very little gain. so I am fixing the type as post processing.

Reply:
It could be one way you could do type checking in place. The type checking also can be done where you post processing (although you might not have as much information about where the offending tokens are).

Reply:
I see we don't have type checks in this case. Below testcase should fail, but it doesn't.
```
DEFINE t1(fp32) = ((20), ('abc'), (1), (10), (5))
corr(t1.col0) = 1::fp64
```

Is it ok if I take this up in another PR?